### PR TITLE
Group generated docs by directive

### PR DESCRIPTION
**Pull request by 2can**
This PR introduces grouping API reference by the `doc` directive in the graphql schema

image

This PR should be merged after the `@doc` directives are added to the schema in saleor - https://github.com/saleor/saleor/pull/12276

After the merging this PR the current folder `docs/api-reference` should be removed and re-generated with the newest schema with `@doc` directives.

### CODE REVIEW

File: .github/workflows/update-reference.yml [hunk 1]
```diff
@@ -6,21 +6,59 @@ on:
...
+
+      - name: Generate schema
+        run: python manage.py get_graphql_schema > schema.graphql
+
+      - name: Upload schema artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: schema
+          path: schema.graphql
+
```
Comment (patrys) [inline .github/workflows/update-reference.yml:36]: We should not do that and, instead, add the schema file to published assets for every core release on GitHub.

**Review by patrys — CHANGES_REQUESTED** [review]

**Review by peelar — APPROVED** [review]

File: .github/workflows/update-reference.yml (hunk 1 above)
Comment (2can) [reply .github/workflows/update-reference.yml:36]: OK, what's the ETA for that?

Comment (maarcingebala) [reply .github/workflows/update-reference.yml:36]: We need to look into modifying our release script to include that, no ETA yet as it wasn't planned.

**Comment by zaiste** [comment]
@2can can you rebase?

**Comment by 2can** [comment]
@zaiste rebased

**Review by patrys — APPROVED** [review]

File: docusaurus.config.js [hunk 2]
```diff
@@ -22,13 +22,18 @@ module.exports = {
     [
       "@graphql-markdown/docusaurus",
       {
-        schema: "http://localhost:8000/graphql/",
+        schema: "./schema.graphql",
```
Comment (korycins) [inline docusaurus.config.js:25]: @2can I think that you should also update the readme:
image

Comment (2can) [reply docusaurus.config.js:25]: @korycins yes, this will be clarified with changes in the release of core which should provide the schema

**Pull request merged**